node/src/GlideClient.ts

```diff
@@ -1020,4 +1022,58 @@ export class GlideClient extends BaseClient {
             decoder: Decoder.String,
         });
     }
+    /**
+     * Incrementally iterate over a collection of keys.
+     * `SCAN` is a cursor based iterator. This means that at every call of the method,
+     * the server returns an updated cursor that the user needs to use as the cursor argument in the next call.
+     * An iteration starts when the cursor is set to "0", and terminates when the cursor returned by the server is "0".
+     *
+     * A full iteration always retrieves all the elements that were present
+     * in the collection from the start to the end of a full iteration.
+     * Elements that were not constantly present in the collection during a full iteration, may be returned or not.
+     *
+     * @see {@link https://valkey.io/commands/scan|valkey.io} for more details.
+     *
+     * @param cursor - The `cursor` used for iteration. For the first iteration, the cursor should be set to "0".
+     * Using a non-zero cursor in the first iteration,
+     * or an invalid cursor at any iteration, will lead to undefined results.
+     * Using the same cursor in multiple iterations will, in case nothing changed between the iterations,
+     * return the same elements multiple times.
+     * If the the db has changed, it may result an undefined behavior.
+     * @param options - (Optional) The options to use for the scan operation.
+     * @param options.match - (Optional) A pattern to match keys against.
+     * @param options.count - (Optional) The number of keys to return per iteration.
+     * The number of keys returned per iteration is not guaranteed to be the same as the count argument.
+     * the argument is used as a hint for the server to know how many "steps" it can use to retrieve the keys.
+     * The default value is 10.
+     * @param options.type - The type of object to scan for see {@link ScanOptions.type} for possible values.
+     * @param options.decoder - (Optional) The decoder to use for the response, see {@link DecoderOption}.
```

Comment:
```suggestion
     * @param options - (Optional) The options to use for the scan operation, see {@link ScanOptions} and {@link DecoderOption}.
```
All rest docs should be in `ScanOptions`

Reply:
Thoughts about it, that's what we did in other commands but not in python for example. I feel it is better UX, are we doing it since It's what we did, or we think it's better?

Reply:
There are 2 reasons:
- an IDE will pop hints from a type when a user open `{` to set options
- doc gen (coming soon) blames incorrect arguments and does not render doc for them

You can format docs as we do for [`ping`](https://github.com/valkey-io/valkey-glide/blob/e4192299e7964c0dfa66753fe7c37281b4870b1e/node/src/GlideClusterClient.ts#L386-L391).
But we apply such style for anonymous types only.